### PR TITLE
Setting created_by/updated_by via header

### PR DESCRIPTION
**Pull request by pacoguzman**
## Description

Provide two new configuration settings:
- `AuthProxyEnabled` to enable retrieving a header to be used on created_by/updated_by
- `AuthProxyHeader` header to use defaults to `X-Email`

## Motivation and Context

Required at https://github.com/checkr/flagr/issues/297

## How Has This Been Tested?

Added unit test to ensure this works as expected

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by pacoguzman** [comment]
We had this changes in place for sometime now and they work just fine. We set up flagr UI behind an `oauth_proxy` instance hat forward the `X-Email` to Flagr

File: pkg/config/env.go [hunk 1]
```diff
@@ -184,6 +184,10 @@ var Config = struct {
 	// "HS256" and "RS256" supported
 	JWTAuthSigningMethod string `env:"FLAGR_JWT_AUTH_SIGNING_METHOD" envDefault:"HS256"`
 
+	// auth proxy in front of flagr to identify user
+	AuthProxyEnabled           bool     `env:"FLAGR_AUTH_PROXY_ENABLED" envDefault:"false"`
```
Comment (zhouzhuojie) [inline pkg/config/env.go]: How about calling it `HeaderAuth`, compatible with `JWTAuth` above. 

```
HeaderAuthEnabled bool
HeaderAuthUserField string
```

File: pkg/handler/subject_test.go [hunk 2]
```diff
@@ -10,9 +11,11 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-func TestGetSubjectFromRequest(t *testing.T) {
+func TestGetSubjectFromJWT(t *testing.T) {
 	var ctx context.Context
 
+	config.Config.JWTAuthEnabled = true
```
Comment (zhouzhuojie) [inline pkg/handler/subject_test.go]: you can also reset the config back for test isolation.

```
defer func() { Config.JWTAuthEnabled = false }()
```

File: pkg/handler/subject_test.go [hunk 3]
```diff
@@ -30,3 +33,16 @@ func TestGetSubjectFromRequest(t *testing.T) {
 	})
 	assert.Equal(t, getSubjectFromRequest(r.WithContext(ctx)), "foo@example.com")
 }
+
+func TestGetSubjectFromOauthProxy(t *testing.T) {
+	var ctx = context.Background()
+
+	config.Config.JWTAuthEnabled = false
+	config.Config.AuthProxyEnabled = true
```
Comment (zhouzhuojie) [inline pkg/handler/subject_test.go]: same here

File: pkg/config/env.go (hunk 1 above)
Comment (pacoguzman) [reply pkg/config/env.go]: Thanks,  sounds better

**Comment by pacoguzman** [comment]
@zhouzhuojie I've applied the feedback ready to review

File: pkg/config/env.go [hunk 4]
```diff
@@ -184,6 +184,10 @@ var Config = struct {
 	// "HS256" and "RS256" supported
 	JWTAuthSigningMethod string `env:"FLAGR_JWT_AUTH_SIGNING_METHOD" envDefault:"HS256"`
 
+	// Identify users through headers
+	HeaderAuthEnabled   bool   `env:"FLAGR_AUTH_PROXY_ENABLED" envDefault:"false"`
```
Comment (zhouzhuojie) [inline pkg/config/env.go]: Oh, the env variables as well

Comment (pacoguzman) [reply pkg/config/env.go]: Opps, done at 942ba8e

**Comment by pacoguzman** [comment]
It seems unit test failed due to other reason https://circleci.com/gh/checkr/flagr/986?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

File: pkg/handler/subject_test.go [hunk 5]
```diff
@@ -30,3 +34,16 @@ func TestGetSubjectFromRequest(t *testing.T) {
...
+func TestGetSubjectFromOauthProxy(t *testing.T) {
+	var ctx = context.Background()
+
+	defer func() { config.Config.HeaderAuthEnabled = false }()
+	config.Config.HeaderAuthEnabled = true
+
+	r, _ := http.NewRequest("GET", "", nil)
+	assert.Equal(t, getSubjectFromRequest(r), "")
+
+	r.Header.Set("X-Email", "foo@example.com")
```
Comment (zhouzhuojie) [inline pkg/handler/subject_test.go]:
```suggestion
	r.Header.Set(config.Config.HeaderAuthUserField, "foo@example.com")
```

**Comment by pacoguzman** [comment]
Thanks, I've applied the suggetion

**Comment by pacoguzman** [comment]
Thanks!

**Pull request merged**